packages/mwp-language-plugin/src/util/getLanguage.js

```diff
@@ -29,7 +29,16 @@ export const getCookieLang: ParseRequestLang = (request: HapiRequest) => {
 export const getUrlLang: ParseRequestLang = (request: HapiRequest) => {
 	const { supportedLangs } = getServerSettings(request);
 	const urlLang = request.url.path.split('/')[1];
-	return supportedLangs.includes(urlLang) && urlLang;
+	const validRequestLang = supportedLangs.includes(urlLang) && urlLang;
+
+	// If request url language is invalid, look at referer for language code
+	// This is for cases when the request url is the api proxy path ie `/mu_api/`
+	if (!validRequestLang && request.headers['referer']) {
```

Comment:
im not familiar with how the referer header gets set, but is it possible for this to be an external URL? if it is, might be worthwhile to only do this if the initial `request.url` contains /mu_api/ in the pattern we expect

Reply:
Yeah I was thinking that - but the proxy endpoint might not necessarily be /mu_api/ because that's a variable you can set in the app config. I'll look into using that variable to check that here

Reply:
Also FYI this function is also called when we do the initial server side render so it is possible in that case that the referrer will be an external url

Reply:
I think the conventional way to do this would be to allow routes to specify plugin-specific configuration - essentially we want this plugin to behave differently on the api proxy route than on the app render route.

We do that for a couple of other plugins in the [mwp-api-proxy-plugin route config](https://github.com/meetup/meetup-web-platform/blob/WP-871_native-app-platformagent/packages/mwp-api-proxy-plugin/src/routes.js#L29) so you can probably follow that pattern. It's not very intuitive from an implementation standpoint because you have to know how to access the route config values from the request (e.g. [like this](https://github.com/meetup/meetup-web-platform/blob/WP-871_native-app-platformagent/packages/mwp-logger-plugin/src/index.js#L20-L22)), but it's not too terrible.

This is definitely stuff that could use an architectural revamp with @jgl2832 at some point